Core/src/org/sleuthkit/autopsy/coreutils/SQLiteTableReader.java

```diff
@@ -0,0 +1,560 @@
+     * @throws IOException            Issue during writing to file.
+     */
+    private void findAndCopySQLiteMetaFile(AbstractFile sqliteFile,
+            String metaFileName) throws NoCurrentCaseException, TskCoreException, IOException {
+
+        Case openCase = Case.getCurrentCaseThrows();
+        SleuthkitCase sleuthkitCase = openCase.getSleuthkitCase();
+        Services services = new Services(sleuthkitCase);
+        FileManager fileManager = services.getFileManager();
+
+        List<AbstractFile> metaFiles = fileManager.findFiles(
+                sqliteFile.getDataSource(), metaFileName,
+                sqliteFile.getParent().getName());
+
+        if (metaFiles != null) {
+            for (AbstractFile metaFile : metaFiles) {
+                copyFileToTempDirectory(metaFile, sqliteFile.getId());
+            }
+        }
+    }
+
+    /**
+     * Copies the file contents into a unique path in the current case temp
+     * directory.
+     *
+     * @param file AbstractFile from the data source
+     * @param id   The input files id value
+     *
+     * @return The path of the file on disk
+     *
+     * @throws IOException            Exception writing file contents
+     * @throws NoCurrentCaseException Current case closed during file copying
+     */
+    private String copyFileToTempDirectory(AbstractFile file, long fileId)
+            throws IOException, NoCurrentCaseException {
+
+        String localDiskPath = Case.getCurrentCaseThrows().getTempDirectory()
+                + File.separator + fileId + file.getName();
+        File localDatabaseFile = new File(localDiskPath);
+        if (!localDatabaseFile.exists()) {
+            ContentUtils.writeToFile(file, localDatabaseFile);
+        }
+        return localDiskPath;
+    }
+
+    /**
+     * Executes the query and assigns resource references to instance variables.
+     *
+     * @param query Input query to execute
+     *
+     * @throws SQLiteTableReaderException
+     */
+    private void openTableResources(String query) throws SQLiteTableReaderException {
+        try {
+            ensureOpen();
+            statement = conn.prepareStatement(query);
+            queryResults = statement.executeQuery();
+            currentMetadata = queryResults.getMetaData();
+            totalColumnCount = currentMetadata.getColumnCount();
+            liveResultSet = true;
+        } catch (SQLException ex) {
+            throw new SQLiteTableReaderException(ex);
+        }
+    }
+
+    /**
+     * Ensures both the statement and the result set for a table are closed.
+     */
+    private void closeTableResources() {
+        try {
+            if (Objects.nonNull(statement)) {
+                statement.close();
+            }
+            if (Objects.nonNull(queryResults)) {
+                queryResults.close();
+            }
+            liveResultSet = false;
+        } catch (SQLException ex) {
+            //Do nothing, can't close.. tried our best.
+        }
```

Comment:
It is fine to not propagate the exception here, but I would prefer that we log it.